src/App.tsx

```diff
@@ -18,56 +55,41 @@
               type="text"
               placeholder="Enter a part of the name"
               className="input"
+              onChange={handleChangeText}
+              value={filterValue}
               data-cy="search-input"
             />
           </div>
 
           <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
             <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
-              </div>
+              {filteredPeople.length !== 0 &&
+                !selectedPerson &&
+                filteredPeople.map(person => (
+                  <div
+                    className="dropdown-item"
+                    data-cy="suggestion-item"
+                    onClick={() => handleSelectedPerson(person)}
+                    key={person.slug}
```

Comment:
Don't generate keys on render. The key prop should be stable between re-renders to prevent unnecessary DOM operations and component state resets. Consider moving the key to the parent component or ensuring it is derived from something that doesn't change on each render.